.github/workflows/check-extra-os.yaml

```diff
@@ -55,11 +55,14 @@ jobs:
           make
           sudo make install
 
-      - uses: r-lib/actions/setup-r-dependencies@v1
+      - uses: r-lib/actions/setup-r-dependencies@v2
         with:
            extra-packages: rcmdcheck
```

Comment:
The syntax has changed a little in v2, see [example in `pkgstats`](https://github.com/ropensci-review-tools/pkgstats/blob/b94328c6f0cc80f85b6b299d4640fd15e78f555b/.github/workflows/check-standard.yaml#L45-L48)

Reply:
The source identifier like any:: and github:: are not strictly necessary but can speed up dependency resolution. But you are right it is good practice to add them, so I will! Though I am not sure why you are using `needs: check` as this would require the `Config/Needs/check` field in the DESCRIPTION.